File: fifth-exercises/python/12.py
# Questão 12: Verifique se um número é primo.

numero = int(input("Digite um número inteiro: "))

if numero <= 1:
    eh_primo = False
elif numero == 2:
    eh_primo = True
elif numero % 2 == 0:
    eh_primo = False
else:
    eh_primo = True
    for i in range(3, int(numero**0.5) + 1, 2):
        if numero % i == 0:
            eh_primo = False
            break

if eh_primo:
    print(f"{numero} é um número primo.")
else:
    print(f"{numero} não é um número primo.")

File: fifth-exercises/c/12.c
#include <stdio.h>

int main() {
    int numero, i;
    int ehPrimo = 1; // Assume que o número é primo

    printf("Digite um número inteiro: ");
    scanf("%d", &numero);

    if (numero <= 1) {
        ehPrimo = 0; // 0 e 1 não são primos
    } else if (numero > 2 && numero % 2 == 0) {
        ehPrimo = 0; // Números pares maiores que 2 não são primos
    } else {
        for (i = 3; i * i <= numero; i += 2) {
            if (numero % i == 0) {
                ehPrimo = 0; // Encontrou um divisor ímpar
                break;
            }
        }
    }

    if (ehPrimo) {
        printf("%d é um número primo.\n", numero);
    } else {
        printf("%d não é um número primo.\n", numero);
    }

    return 0;
}
